pydantic_settings/sources.py

```diff
@@ -449,12 +449,7 @@ def __init__(
         super().__init__(settings_cls, env_nested_delimiter, env_prefix_len)
 
     def _load_env_vars(self) -> Mapping[str, Optional[str]]:
-        env_vars = super()._load_env_vars()
-        dotenv_vars = self._read_env_files(self.settings_cls.model_config.get('case_sensitive', False))
-        if dotenv_vars:
-            env_vars = {**dotenv_vars, **env_vars}
-
-        return env_vars
+        return self._read_env_files(self.settings_cls.model_config.get('case_sensitive', False))
```

Comment:
I'm a bit surprised that `DotEnvSettingsSource` inherits from `EnvSettingsSource` - are you sure that's how it should be?

It might be cleaner if they were separate sources and we use the normal logic to merge them?

Reply:
Actually, the env settings source and dotenv setting source were in the same class and all the logic was the same in `V1`(EnvSettingsSource). There was only one logic related to dotenv, loading env variables from dotenv file.

There was some logic like `explode_env_vars` which has to be done on both sources.
Actually, dotenv is not a separate source, it's a just some functionality for loading env variables.

We can:

- Keep these classes as is
- Merge these two classes into one class
- Separate these two classes as two separate sources and it will introduce a breaking change

Reply:
okay, sounds good.